sync/tryrwmutex.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+)
+
+const (
+	readOffset  = uint64(1)
+	writeOffset = uint64(1 << 20)
+	tryOffset   = uint64(1 << 40)
+
+	tryMask = uint64(1099511627775) // equivalent to setting the first 39 bits to '1'.
```

Comment:
hex is better for this purpose. This is actually FFFFFFFFFF, whereas the comment states that it is 7FFFFFFFFF. You could also use `(1<<39)-1`.
It seems like this code works fine as-is, though.